pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -982,9 +982,7 @@ func (m *kubeGenericRuntimeManager) computePodActions(ctx context.Context, pod *
 		var reason containerKillReason
 		restart := shouldRestartOnFailure(pod)
 		// Do not restart if only the Resources field has changed with InPlacePodVerticalScaling enabled
-		if _, _, changed := containerChanged(&container, containerStatus); changed &&
-			(!isInPlacePodVerticalScalingAllowed(pod) ||
-				kubecontainer.HashContainerWithoutResources(&container) != containerStatus.HashWithoutResources) {
+		if _, _, changed := containerChanged(&container, containerStatus); changed {
```

Comment:
Does this properly handle the case of a Pod's UID appearig to change?  For example, we miss an event (or are down while it happens) which deletes the Pod and recreates it with the same name, but slightly different config?  This has happened in ther places where the object was supposed to be immutable but people find ways around it, on purpose or otherwise.

Reply:
From the defensive programming perspective, we seem to have to do this.

Reply:
We don't need to factor in pod UID here.

The containerStatus we're comparing against comes from podStatus, which is looked up by pod UID.

If the pod uid changes, we won't even have a containerStatus to compare against.

Reply:
Great!  Maybe we can drop a comment there so that future readers know this?

Reply:
Comment added